### ISSUE TITLE
Outline (Table of Contents) doesn't show roman page number.

### ISSUE THREAD

**Issue by nafkhanzam** [typst/typst#488]
I changed the first 2 pages with roman page numbering, then reset it to default numbering on the 3rd page, but the table of contents page is still showing the default numbering.

```typst
set page(
  paper: "a4",
  margin: (
    left: 4cm,
    right: 3cm,
    y: 3cm,
  ),
  numbering: "i",
)

...

heading(numbering: none)[DAFTAR ISI]
outline(title: none, indent: true)
pagebreak()

...

set page(numbering: "1")
counter(page).update(1)

...
```

![image](https://user-images.githubusercontent.com/48153421/229215600-04bd7717-b9ab-42d1-8fa6-035bdcf1ac4d.png)

---

**Current workaround**

```typst
#let outline(title: "Contents", depth: none, indent: false) = {
  heading(title, numbering: none)
  locate(it => {
    let elements = query(heading, after: it)

    for i, el in elements {
      if depth != none and el.level > depth { continue }

      let maybe_number = if el.numbering != none {
        numbering(el.numbering, ..counter(heading).at(el.location()))
        " "
      }
      let line = {
        if indent {
          h(1em * (el.level - 1 ))
        }

        if el.level == 1 {
          v(weak: true, 0.5em)
          set text(weight: "bold")
          maybe_number
          el.body
        } else {
          maybe_number
          el.body
        }

        // Filler dots
        box(width: 1fr, h(3pt) + box(width: 1fr, repeat[.]) + h(3pt))
        // Page number
        let page-number = counter(page).at(el.location()).first()
        if i == 0 {
          page-number = numbering("i", page-number)
        }
        str(page-number)

        linebreak()
      }

      link(el.location(), line)
    }
  })
}

#set page(
  paper: "a4",
  margin: (
    left: 4cm,
    right: 3cm,
    y: 3cm,
  ),
  numbering: "i",
)

#heading(numbering: none)[DAFTAR ISI]
#outline(title: none, indent: true)
#pagebreak()

= ROMAN

#set page(numbering: "1")
#counter(page).update(1)

= oiih
== usaiduhasd

=== asisodhf
#pagebreak()
= Head
```

![image](https://user-images.githubusercontent.com/48153421/229215799-cb961fe9-f4e9-46a7-803a-d675cd1549b8.png)

**Comment by CreatorSiSo** [comment]
The problem is that the outline cant apply the numbering argument of page, because it only has access to the page counter.
https://github.com/typst/typst/blob/cc67e5309994eefd1bd1c892f040ec7912052a4b/library/src/meta/outline.rs#L163-L166

**Comment by Dherse** [comment]
This should be fixed by #594.

**Comment by nafkhanzam** [comment]
> This should be fixed by #594.

Confirmed and tested. Thanks!
